### PR TITLE
Sample python bilinear initializer at integral points in y-direction

### PR DESCRIPTION
**Pull request by vladoovtcharov**
Bilinear initializer was sampling at fractional points in y rather then sampling at integral points (like is done for x)

### CODE REVIEW

**Comment by ankkhedia** [comment]
@vladoovtcharov Thanks for your contribution!

@mxnet-label-bot [pr-awaiting-review, Python]

@vandanavk @Roshrini @sandeep-krishnamurthy Could you please help review the PR

File: python/mxnet/initializer.py [hunk 1]
```diff
@@ -656,7 +656,7 @@ def _init_weight(self, _, arr):
         c = (2 * f - 1 - f % 2) / (2. * f)
         for i in range(np.prod(shape)):
             x = i % shape[3]
-            y = (i / shape[3]) % shape[2]
+            y = (i // shape[3]) % shape[2]
```
Comment (anirudhacharya) [inline python/mxnet/initializer.py]: there is a `_init_bilinear` method in the base Initializer class - https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/initializer.py#L212 . Is this class redundant? 

Also I see that many of the initializers do not have tests. Could you please add a test for this change by creating a new file `test_initializer.py` in the tests folder. Tests for other initializers can be added to the same file later.

**Comment by ankkhedia** [comment]
@sandeep-krishnamurthy Could you please label this issue as pr-awaiting-response

**Comment by kalyc** [comment]
Hi @vladoovtcharov thanks for your contribution! Could you please address the comments above?

**Comment by stu1130** [comment]
@vladoovtcharov ping again.
let us know if you need any help!

**Comment by vladoovtcharov** [comment]
`_init_bilinear` looks redundant and also seems to be deprecated from the comments but has the same issue.
Some initializers have tests in `incubator-mxnet/tests/python/unittest/test_init.py` but not bilinear I'm not sure if I'll have time to add one for a bit

**Comment by anirudhacharya** [comment]
The issue of code redundancy still remains. I am not sure about the history about this bit of code, pinging @szha to maybe help shed some light here.

**Comment by vandanavk** [comment]
what are the next steps on this PR @vladoovtcharov @anirudhacharya @szha ?

**Comment by vladoovtcharov** [comment]
python2 test seems to be failing (but works for python 3). I don't currently have python 2 installed but I'll try to test out soon

**Comment by vladoovtcharov** [comment]
Wasn't handling division correctly in python 2. Seems to be working now

**Comment by Roshrini** [comment]
@sandeep-krishnamurthy Can you please review/merge this PR?
@mxnet-label-bot Update [Python, pr-awaiting-merge]

**Comment by anirudhacharya** [comment]
@Roshrini can you please merge this

**Pull request merged**